### PR TITLE
Add printf builtin and support for basic escape sequences

### PR DESCRIPTION
**Pull request by prologic**

### CODE REVIEW

**Comment by prologic** [comment]
Need to do something similar to what I've done in monkey-lang:

=> https://git.mills.io/prologic/monkey-lang/src/commit/d5ae2761ee7f6e332340cf700861f4a6bbf03a8b/lexer/lexer.go#L243-L272

**Comment by prologic** [comment]
Been looking for a library that deals with encoding escape sequences into their proper form, but haven't found anything anywhere 😅 I really hate writing code that should already exist damnit 😂

**Comment by NicoNex** [comment]
The good side is that we keep Tau dependency-free 🤣 .

**Comment by NicoNex** [comment]
We need to fix the case where the user inputs a string like `"example\"` since the lexer goes into an infinite loop.
Additionally we need to add support for raw strings probably with a `ast.RawString` object.

File: internal/lexer/lexer.go [hunk 1]
```diff
@@ -155,6 +155,17 @@ func lexString(l *lexer) stateFn {
...
+	if l.peek() == '`' {
+		l.emit(item.RawString)
+		l.next()
+		l.ignore()
+		return lexExpression
+	}
+	l.next()
+	return lexRawString
+}
+
```
Comment (prologic) [inline internal/lexer/lexer.go:168]: Nice 👌

**Comment by NicoNex** [comment]
For the printf the situation is trickier than I initially thought since the placeholders expect primitive types not Tau types, so for example `%d` with `*obj.Integer` will print the address of the object.

**Comment by NicoNex** [comment]
I don't think we still need the printf since I just implemented python-style [string interpolation](https://github.com/NicoNex/tau#string).

**Comment by prologic** [comment]
Sounds okay to me 👌

**Pull request closed without merging**